water/reverse.go

```diff
@@ -0,0 +1,94 @@
+package water
+
+import (
+	"context"
+	"encoding/base64"
+	"net"
+	"sync"
+
+	"github.com/getlantern/golog"
+	"github.com/refraction-networking/water"
+	_ "github.com/refraction-networking/water/transport/v0"
+)
+
+var log = golog.LoggerFor("water")
+
+type llistener struct {
+	net.Listener
+
+	connections  chan net.Conn
+	closedSignal chan struct{}
+	closeOnce    sync.Once
+	closeError   error
+}
+
+func NewReverseListener(ctx context.Context, address string, wasm string) (*llistener, error) {
```

Comment:
I'm confused about the "reverse" naming here and elsewhere.  Is this referring to the reverse transport I suggested testing with [here](https://github.com/getlantern/engineering/issues/1437#issuecomment-2158981893)?  If so, I think we can do away with all of the "reverse" parts of these names - these constructs are all generic and will work with any WASM (as they should!).

Alternatively, is "reverse" here used in the same manner as "reverse proxy"?  If so, again I think we can just drop the "reverse" parts of these names.

Reply:
hmmmmmm I've added as reverse because I thought it would only work for the reverse WASM. I'll remove that